### PR TITLE
Add support for custom plugin code injection

### PR DESCRIPTION
**Pull request by virgilsky**
Fixes [GH-269](https://github.com/ascopes/protobuf-maven-plugin/issues/269)

### CODE REVIEW

**Comment by ascopes** [comment]
Sounds good to me. Is it possible to include an integration test that calls such a plugin to verify it works as expected and doesn't regress in the future?

If not, happy to merge as is!

File: protobuf-maven-plugin/src/main/java/io/github/ascopes/protobufmavenplugin/generation/SourceCodeGenerator.java [hunk 1]
```diff
@@ -129,6 +128,9 @@ public boolean generate(GenerationRequest request) throws ResolutionException, I
             request.isLiteEnabled()
         ));
 
+    // Add the plugins after the enabled languages to support generated code injection
```
Comment (ascopes) [inline protobuf-maven-plugin/src/main/java/io/github/ascopes/protobufmavenplugin/generation/SourceCodeGenerator.java:131]:
```suggestion
    // GH-269: Add the plugins after the enabled languages to support generated code injection
```

Comment (virgilsky) [reply protobuf-maven-plugin/src/main/java/io/github/ascopes/protobufmavenplugin/generation/SourceCodeGenerator.java]: Agree. Updated

**Comment by virgilsky** [comment]
> Sounds good to me. Is it possible to include an integration test that calls such a plugin to verify it works as expected and doesn't regress in the future?
> 
> If not, happy to merge as is!

The plugin I mentioned is not public. If it is really needed, I could write and publish a mock one just for this.

**Comment by ascopes** [comment]
If it is any real effort on your side to do, then not to worry! Happy to merge as is for now.

I will eventually get around to writing the lower level unit tests for this stuff which will cover behaviours like this in a more granular way than the integration tests can currently handle.

I'll get this merged for now. Is there any urgency to this being released? I did a release a couple of hours ago but I can kick another off if it is blocking you at all.

**Comment by virgilsky** [comment]
I would appreciate if you can merge it now, however I will raise another issue regarding resolving plugins in PATH. After that one, I would greatly appreciate a release.

**Pull request merged**